### ISSUE TITLE
[BUG] Error in `collect()`:

### ISSUE THREAD

**Issue by ndaniel** [LieberInstitute/recount3#39]
When running this:

```
library(recount3)
recount3::create_rse_manual(
    project = "SRP186687",
    project_home = "data_sources/sra",
    organism = "human",
    annotation = "gencode_v26",
    type = "gene"
)
```
this error shows up:
```
2023-11-10 19:11:54.627105 downloading and reading the metadata.
Error in `collect()`:
! Failed to collect lazy table.
Caused by error in `db_collect()`:
! Arguments in `...` must be used.
✖ Problematic argument:
• ..1 = Inf
 Did you misspell an argument name?
Run `rlang::last_trace()` to see where the error occurred.
```

More info:
```
> sessionInfo()
R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin22.6.0 (64-bit)
Running under: macOS Ventura 13.6.2

Matrix products: default
BLAS:   /opt/homebrew/Cellar/openblas/0.3.24/lib/libopenblasp-r0.3.24.dylib 
LAPACK: /opt/homebrew/Cellar/r/4.3.2/lib/R/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Helsinki
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] recount3_1.10.2             SummarizedExperiment_1.30.2
 [3] Biobase_2.60.0              GenomicRanges_1.52.1       
 [5] GenomeInfoDb_1.36.4         IRanges_2.34.1             
 [7] S4Vectors_0.38.2            BiocGenerics_0.46.0        
 [9] MatrixGenerics_1.12.3       matrixStats_1.1.0          

loaded via a namespace (and not attached):
 [1] rjson_0.2.21             lattice_0.22-5           vctrs_0.6.4             
 [4] tools_4.3.2              bitops_1.0-7             generics_0.1.3          
 [7] curl_5.1.0               parallel_4.3.2           tibble_3.2.1            
[10] fansi_1.0.5              RSQLite_2.3.3            blob_1.2.4              
[13] pkgconfig_2.0.3          R.oo_1.25.0              Matrix_1.6-2            
[16] data.table_1.14.8        dbplyr_2.4.0             lifecycle_1.0.4         
[19] GenomeInfoDbData_1.2.10  compiler_4.3.2           Rsamtools_2.16.0        
[22] Biostrings_2.68.1        codetools_0.2-19         RCurl_1.98-1.13         
[25] yaml_2.3.7               pillar_1.9.0             crayon_1.5.2            
[28] R.utils_2.12.2           BiocParallel_1.34.2      DelayedArray_0.26.7     
[31] cachem_1.0.8             sessioninfo_1.2.2        abind_1.4-5             
[34] tidyselect_1.2.0         purrr_1.0.2              dplyr_1.1.3             
[37] restfulr_0.0.15          fastmap_1.1.1            grid_4.3.2              
[40] cli_3.6.1                magrittr_2.0.3           S4Arrays_1.0.6          
[43] XML_3.99-0.15            utf8_1.2.4               filelock_1.0.2          
[46] bit64_4.0.5              XVector_0.40.0           httr_1.4.7              
[49] bit_4.0.5                R.methodsS3_1.8.2        memoise_2.0.1           
[52] BiocIO_1.10.0            BiocFileCache_2.8.0      rtracklayer_1.60.1      
[55] rlang_1.1.2              glue_1.6.2               DBI_1.1.3               
[58] R6_2.5.1                 GenomicAlignments_1.36.0 zlibbioc_1.46.0 
```

**Comment by sdhutchins** [comment]
I'm also getting the same error.

```console
R version 4.3.1 (2023-06-16)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.0

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] decoupleR_2.6.0             DESeq2_1.40.2               data.table_1.14.8           recount3_1.10.2            
 [5] SummarizedExperiment_1.30.2 Biobase_2.60.0              GenomicRanges_1.52.1        GenomeInfoDb_1.36.4        
 [9] IRanges_2.34.1              S4Vectors_0.38.2            BiocGenerics_0.46.0         MatrixGenerics_1.12.3      
[13] matrixStats_1.1.0           lubridate_1.9.3             forcats_1.0.0               stringr_1.5.0              
[17] dplyr_1.1.3                 purrr_1.0.2                 readr_2.1.4                 tidyr_1.3.0                
[21] tibble_3.2.1                ggplot2_3.4.4               tidyverse_2.0.0            

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0         blob_1.2.4               filelock_1.0.2           R.utils_2.12.2          
 [5] Biostrings_2.68.1        bitops_1.0-7             fastmap_1.1.1            RCurl_1.98-1.13         
 [9] BiocFileCache_2.8.0      GenomicAlignments_1.36.0 XML_3.99-0.15            digest_0.6.33           
[13] timechange_0.2.0         lifecycle_1.0.4          RSQLite_2.3.3            magrittr_2.0.3          
[17] compiler_4.3.1           rlang_1.1.2              tools_4.3.1              utf8_1.2.4              
[21] yaml_2.3.7               rtracklayer_1.60.1       knitr_1.45               S4Arrays_1.0.6          
[25] bit_4.0.5                curl_5.1.0               DelayedArray_0.26.7      BiocParallel_1.34.2     
[29] abind_1.4-5              withr_2.5.2              R.oo_1.25.0              grid_4.3.1              
[33] fansi_1.0.5              colorspace_2.1-0         scales_1.2.1             cli_3.6.1               
[37] rmarkdown_2.25           crayon_1.5.2             generics_0.1.3           rstudioapi_0.15.0       
[41] httr_1.4.7               tzdb_0.4.0               rjson_0.2.21             sessioninfo_1.2.2       
[45] DBI_1.1.3                cachem_1.0.8             zlibbioc_1.46.0          parallel_4.3.1          
[49] XVector_0.40.0           restfulr_0.0.15          vctrs_0.6.4              Matrix_1.6-2            
[53] hms_1.1.3                bit64_4.0.5              locfit_1.5-9.8           glue_1.6.2              
[57] codetools_0.2-19         stringi_1.7.12           gtable_0.3.4             BiocIO_1.10.0           
[61] munsell_0.5.0            pillar_1.9.0             htmltools_0.5.7          GenomeInfoDbData_1.2.10 
[65] R6_2.5.1                 dbplyr_2.4.0             evaluate_0.23            lattice_0.22-5          
[69] Rsamtools_2.16.0         R.methodsS3_1.8.2        memoise_2.0.1            Rcpp_1.0.11             
[73] xfun_0.41                pkgconfig_2.0.3     

```

**Comment by ndaniel** [comment]
Another example on another computer where it does **not** work.
```
> sessionInfo()
R version 4.3.2 (2023-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=fi_FI.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=fi_FI.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=fi_FI.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=fi_FI.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/Helsinki
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] recount3_1.10.2             SummarizedExperiment_1.30.2
 [3] Biobase_2.60.0              GenomicRanges_1.52.1       
 [5] GenomeInfoDb_1.36.4         IRanges_2.34.1             
 [7] S4Vectors_0.38.2            BiocGenerics_0.46.0        
 [9] MatrixGenerics_1.12.3       matrixStats_1.1.0          

loaded via a namespace (and not attached):
 [1] rjson_0.2.21             lattice_0.22-5           vctrs_0.6.4             
 [4] tools_4.3.2              bitops_1.0-7             generics_0.1.3          
 [7] curl_5.1.0               parallel_4.3.2           tibble_3.2.1            
[10] fansi_1.0.5              RSQLite_2.3.3            blob_1.2.4              
[13] pkgconfig_2.0.3          R.oo_1.25.0              Matrix_1.6-2            
[16] data.table_1.14.8        dbplyr_2.4.0             lifecycle_1.0.4         
[19] GenomeInfoDbData_1.2.10  compiler_4.3.2           Rsamtools_2.16.0        
[22] Biostrings_2.68.1        codetools_0.2-19         RCurl_1.98-1.13         
[25] yaml_2.3.7               pillar_1.9.0             crayon_1.5.2            
[28] R.utils_2.12.2           BiocParallel_1.34.2      DelayedArray_0.26.7     
[31] cachem_1.0.8             sessioninfo_1.2.2        abind_1.4-5             
[34] tidyselect_1.2.0         purrr_1.0.2              dplyr_1.1.3             
[37] restfulr_0.0.15          fastmap_1.1.1            grid_4.3.2              
[40] cli_3.6.1                magrittr_2.0.3           S4Arrays_1.0.6          
[43] XML_3.99-0.15            utf8_1.2.4               filelock_1.0.2          
[46] bit64_4.0.5              XVector_0.40.0           httr_1.4.7              
[49] bit_4.0.5                R.methodsS3_1.8.2        memoise_2.0.1           
[52] BiocIO_1.10.0            BiocFileCache_2.8.0      rtracklayer_1.60.1      
[55] rlang_1.1.2              glue_1.6.2               DBI_1.1.3               
[58] R6_2.5.1                 GenomicAlignments_1.36.0 zlibbioc_1.46.0         

```

**Comment by ndaniel** [comment]
and on a third computer **it works**:
```
R version 4.3.2 (2023-10-31)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Monterey 12.6.2

Matrix products: default
BLAS:  /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
time zone: Europe/Helsinki
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages: 
[1] ggsci_3.0.0                 lubridate_1.9.3             forcats_1.0.0               
[4] stringr_1.5.0               dplyr_1.1.3                 purrr_1.0.2                 
[7] tidyr_1.3.0                 tibble_3.2.1                tidyverse_2.0.0            
[10] data.table_1.14.8           readr_2.1.4                 recount3_1.12.0            
[13] biomaRt_2.58.0              rgl_1.2.1                   DESeq2_1.42.0              
[16] SummarizedExperiment_1.32.0 Biobase_2.62.0              MatrixGenerics_1.14.0      
[19] matrixStats_1.1.0           GenomicRanges_1.54.1        GenomeInfoDb_1.38.1        
[22] IRanges_2.36.0              S4Vectors_0.40.1            BiocGenerics_0.48.1        
[25] edgeR_4.0.1                 limma_3.58.1                fgsea_1.28.0               
[28] ggplot2_3.4.4              

loaded via a namespace (and not attached): 
[1] DBI_1.1.3                bitops_1.0-7             rlang_1.1.2              
[4] magrittr_2.0.3           compiler_4.3.2           RSQLite_2.3.3            
[7] png_0.1-8                vctrs_0.6.4              pkgconfig_2.0.3         
[10] crayon_1.5.2             fastmap_1.1.1            dbplyr_2.4.0            
[13] XVector_0.42.0           utf8_1.2.4               Rsamtools_2.18.0        
[16] rmarkdown_2.25           tzdb_0.4.0               sessioninfo_1.2.2       
[19] bit_4.0.5                xfun_0.41                zlibbioc_1.48.0         
[22] cachem_1.0.8             jsonlite_1.8.7           progress_1.2.2          
[25] blob_1.2.4               DelayedArray_0.28.0      BiocParallel_1.36.0     
[28] parallel_4.3.2           prettyunits_1.2.0        R6_2.5.1                
[31] stringi_1.7.12           rtracklayer_1.62.0       extrafontdb_1.0         
[34] Rcpp_1.0.11              knitr_1.45               base64enc_0.1-3         
[37] R.utils_2.12.2           extrafont_0.19           timechange_0.2.0        
[40] Matrix_1.6-2             tidyselect_1.2.0         rstudioapi_0.15.0       
[43] abind_1.4-5              yaml_2.3.7               codetools_0.2-19        
[46] curl_5.1.0               lattice_0.22-5           withr_2.5.2             
[49] KEGGREST_1.42.0          evaluate_0.23            BiocFileCache_2.10.1    
[52] xml2_1.3.5               Biostrings_2.70.1        pillar_1.9.0            
[55] filelock_1.0.2           generics_0.1.3           RCurl_1.98-1.13         
[58] hms_1.1.3                munsell_0.5.0            scales_1.2.1            
[61] glue_1.6.2               tools_4.3.2              BiocIO_1.12.0           
[64] locfit_1.5-9.8           GenomicAlignments_1.38.0 XML_3.99-0.15           
[67] fastmatch_1.1-4          cowplot_1.1.1            grid_4.3.2              
[70] Rttf2pt1_1.3.12          AnnotationDbi_1.64.1     colorspace_2.1-0        
[73] GenomeInfoDbData_1.2.11  restfulr_0.0.15          cli_3.6.1               
[76] rappdirs_0.3.3           fansi_1.0.5              S4Arrays_1.2.0          
[79] gtable_0.3.4             R.methodsS3_1.8.2        digest_0.6.33           
[82] SparseArray_1.2.2        htmlwidgets_1.6.2        rjson_0.2.21            
[85] memoise_2.0.1            htmltools_0.5.7          R.oo_1.25.0             
[88] lifecycle_1.0.4          httr_1.4.7               statmod_1.5.0           
[91] bit64_4.0.5            
```

**Comment by ndaniel** [comment]
It looks like this bug is linked to https://github.com/Bioconductor/BiocFileCache/pull/50 and downgrading *dbplyr* to an older version (from 2.4.0 to 2.3.4) is a temporary workaround.
```
install.packages("devtools")
devtools::install_version("dbplyr", version = "2.3.4")
```

**Comment by sdhutchins** [comment]
I had success with the workaround as well.

**Comment by lcolladotor** [comment]
Thanks for using `recount3`, reporting this issue, and finding the related problem. From https://github.com/Bioconductor/BiocFileCache/pull/50#issuecomment-1781113054, my understanding is that this issue has been fixed in bioc release (3.18) and devel (3.19).  

Can you confirm that this is the case? If so, we can then close this issue.

Best,
Leo
